File: Leetcode_solutions/Longest Substring Without Repeating Characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
  const n = s.length;
  let ans = 0;
  const map = new Map();
  for (let j = 0, i = 0; j < n; j++) {
    if (map.has(s.charAt(j))) {
      i = Math.max(map.get(s.charAt(j)), i);
    }
    ans = Math.max(ans, j - i + 1);
    map.set(s.charAt(j), j + 1);
  }
  return ans;
};

File: Leetcode_solutions/Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s)<=1:
            return len(s)
        ans=0
        start=0
        for i in range(len(s)):
            if len(s[start:i+1]) == len(set(s[start:i+1])):
                ans = max(ans, len(s[start:i+1]))
            else:
                start+=1
        return ans